packages/components/src/DataList/DataList.css

```diff
@@ -25,3 +25,25 @@
   padding: var(--space-small);
   border-bottom: var(--border-base) solid var(--color-border);
 }
+
+.layout {
+  @media (--small-screens-and-below) {
+    display: var(--data-list-xs-display);
+  }
+
+  @media (--small-screens-and-up) {
+    display: var(--data-list-sm-display);
+  }
+
+  @media (--medium-screens-and-up) {
+    display: var(--data-list-md-display);
+  }
+
+  @media (--large-screens-and-up) {
+    display: var(--data-list-lg-display);
+  }
+
+  @media (--extra-large-screens-and-up) {
+    display: var(--data-list-xl-display);
+  }
+}
```

Comment:
I wouldn't use CSS to deal with this. 

You're rendering twice (or 5, if you hit all breakpoints) the amount of elements and functions on the page. CSS is hiding them and preventing it from being painted, but the functions and side effects that comes with those components still exists and are running.

Reply:
I initially was conditionally rendering this using a `useMediaQueries` hook. I can update it to use that

Reply:
@darryltec I am doing something to update this, however this is already happening with GridCells so I am not sure if it something to address at the moment.

Reply:
> however this is already happening with GridCells so I am not sure if it something to address at the moment.

usage is a bit different no? for grid, you can't really hide the element

Reply:
@darryltec I updated it.